arangod/RestHandler/RestQueryHandler.cpp

```diff
@@ -80,6 +92,105 @@ RestStatus RestQueryHandler::execute() {
   return RestStatus::DONE;
 }
 
+void RestQueryHandler::dumpQueryRegistry() {
+  if (!ExecContext::current().isSuperuser()) {
+    generateError(rest::ResponseCode::FORBIDDEN, TRI_ERROR_FORBIDDEN);
+    return;
+  }
+
+  if (!_vocbase.server().getFeature<QueryRegistryFeature>().enableDebugApis()) {
+    // debug API turned off
+    generateError(rest::ResponseCode::FORBIDDEN, TRI_ERROR_FORBIDDEN);
+    return;
+  }
+
+  bool const fanout = ServerState::instance()->isCoordinator() &&
+                      !_request->parsedValue("local", false);
+
+  VPackBuilder builder;
+  builder.openObject();
+  builder.add("queries", VPackValue(VPackValueType::Array));
+
+  if (fanout) {
+    TRI_ASSERT(ServerState::instance()->isCoordinator());
+    auto& ci = _vocbase.server().getFeature<ClusterFeature>().clusterInfo();
+
+    NetworkFeature const& nf = _vocbase.server().getFeature<NetworkFeature>();
+    network::ConnectionPool* pool = nf.pool();
+    if (pool == nullptr) {
+      THROW_ARANGO_EXCEPTION(TRI_ERROR_SHUTTING_DOWN);
+    }
+
+    std::vector<network::FutureRes> futures;
+    auto auth = AuthenticationFeature::instance();
+
+    network::RequestOptions options;
+    options.database = _vocbase.name();
+    options.timeout = network::Timeout(30.0);
+    options.param("local", "true");
+
+    VPackBuffer<uint8_t> body;
+
+    auto servers = ci.getCurrentDBServers();
+
+    for (auto const& server : servers) {
+      if (server == ServerState::instance()->getId()) {
+        // ourselves!
+        continue;
+      }
+
+      network::Headers headers;
+      if (auth != nullptr && auth->isActive()) {
+        auto const& username = ExecContext::current().user();
+
+        if (!username.empty()) {
+          headers.try_emplace(
+              StaticStrings::Authorization,
+              "bearer " + fuerte::jwt::generateUserToken(
+                              auth->tokenCache().jwtSecret(), username));
+        } else {
+          headers.try_emplace(StaticStrings::Authorization,
+                              "bearer " + auth->tokenCache().jwtToken());
+        }
+      }
+
+      auto f = network::sendRequestRetry(
+          pool, "server:" + server, fuerte::RestVerb::Get,
+          "/_api/query/registry", body, options, std::move(headers));
+      futures.emplace_back(std::move(f));
+    }
+
+    if (!futures.empty()) {
+      auto responses = futures::collectAll(futures).get();
```

Comment:
We actively try to get rid of blocking code and it was never easier to write non-blocking code using coroutines. 
I will provide a PR that makes writing async rest handler more easily.